Lib/pathlib/_abc.py

```diff
@@ -450,13 +450,29 @@ def match(self, path_pattern, *, case_sensitive=None):
         if case_sensitive is None:
             case_sensitive = _is_case_sensitive(self.pathmod)
         sep = path_pattern.pathmod.sep
-        if path_pattern.anchor:
-            pattern_str = str(path_pattern)
-        elif path_pattern.parts:
-            pattern_str = str('**' / path_pattern)
-        else:
+        our_parts = self.parts[::-1]
+        pat_parts = path_pattern.parts[::-1]
```

Comment:
Use full word so no need to guess about later uses?
```suggestion
        pattern_parts = path_pattern.parts[::-1]
```